lib/translator/inquiry.cc

```diff
@@ -185,6 +185,88 @@ StatusCode InquiryToNvme(absl::Span<const uint8_t> raw_scsi,
   return StatusCode::kSuccess;
 }
 
+void TranslateLogicalBlockProvisioningVpd(
+    const nvme::IdentifyControllerData& identify_ctrl,
+    const nvme::IdentifyNamespace& identify_ns, absl::Span<uint8_t> buffer) {
+  bool ad = identify_ctrl.oncs.dsm;
+
+  scsi::LogicalBlockProvisioningVpd result = {
+      .page_code = scsi::PageCode::kLogicalBlockProvisioningVpd,
+      .page_length = 0x04,  // TODO: constant val in common.h
+
+      // THRESHOLD_EXPONENT
+      // Shall be set to 00h to indicate that there are no thin provisioning
+      // thresholds This would require modification if thin-provisioning is
+      // supported
+
+      // TODO:
+      // Right now we assume this field is 0.
+      // If thin provisioning is enabled, we can calculate a valid threshold
+      // exponnet from the READ CAPACITY (16) parameter data.
+
+      // .threshold_exponent= identify_ns.nsfeat.thin_prov ? read_capacity : 0,
+
+      // LBPRZ
+      // Shall be set to 1 if Dataset Management command – Deallocate (AD)
+      // attribute is supported and the device returns all zeros for reads of
+      // deallocated LBAs, otherwise set to 0.
+
+      // TODO: the AD attribute itself may not guarantee the device returns all
+      // zeros for reads of deallocated LBAs. May have to double check this one
+      // if there are any errors.
+      .lbprz = ad,
+
+      // ANC_SUP
+      // Shall be set to 0, to indicate that setting the ANCHOR bit in UNMAP is
+      // not supported if the namespace is not resource or thin provisioned.
+
+      // Shall be set to 1, to indicate that setting the ANCHOR bit in UNMAP is
+      // supported if the namespace is resource provisioned.
+
+      // May be set to 1 if the namespace is thin provisioned.
+
+      // TODO: Lack of support from thin provisioning by itself may not
+      // guarantee lack of resource provisioning. There may be an explicit flag
+      // about resource provisioning.
+
+      .anc_sup = identify_ns.nsfeat.thin_prov,
```

Comment:
per @w-michal we are not currently supporting the anchor bit in unmap, so let's just always return 0 here.